### PR TITLE
Support Order Details deep link

### LINKED ISSUE #7323: Handle Order Details Universal Link
## Description

copied from https://github.com/woocommerce/woocommerce-ios/issues/7537

With this task we want to handle order details universal links and navigate the user to the details of the order specified by the parameters in the link URL.
The path schema would be `/orders/details?blog_id=<blog id number>&order_id=<order id number>` e.g /orders/details?blog_id=1&order_id=17 would navigate to the orders tab and then open the details for the `order_id` 17, of site with `blog_id` 1
The main challenge, therefore, is to change the site of the app if the passed blog id is not the same as the current app site id.

### PR DESCRIPTION
**Pull request by wzieba**
Closes: #7323

### Description

This PR adds support for deep links for opening orders details.

### Testing instructions

Please perform tests on the release variant (`vanillaRelease`). It's important because you'll test domain verification as well.

1. Prepare a few links:
```
1 https://woocommerce.com/mobile/orders/details?blog_id=<blog_A_id>&order_id=<order_A_id>

2 https://woocommerce.com/mobile/orders/details?blog_id=<blog_A_id>&order_id=<order_B_id>

3 https://woocommerce.com/mobile/orders/details?blog_id=<blog_B>&order_id=<order_C_id>

4 https://woocommerce.com/mobile/orders/details?blog_id=<random_numbers>&order_id=<random_numbers>

Not valid /*5 https://woocommerce.com/mobile/orders/details?blog_id=<blog_A_id>&order_id=<random_numbers>*/

6 https://woocommerce.com/mobile/orders/details?order_id=<random_numbers>

7 https://woocommerce.com/mobile/orders/details?blog_id=<random_numbers>
```

1. Click on the 1st link. Assert you can see `order A` on `blog A`
2. Click on the 2nd link. Assert you can see `order B` on `blog A`
3. Click on the 3rd link. Assert you can see `order C` on `blog B`. Then repeat test number 2. Make sure the app changes blogs correctly.
4. Click on the 4th link. Assert that the app just opened on the stats screen
5. Click on the 5th link. Assert that the app just opened on the stats screen
6. Click on the 6th link. Assert that the app just opened on the stats screen
7. Click on the 7th link. Assert that the app just opened on the stats screen

### Images/gif

https://user-images.githubusercontent.com/5845095/193794365-494f9e5f-e540-400a-a761-ad91306568d5.mov

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by nbradbury** [comment]
@wzieba I'm seeing some unexpected behavior here. I tried the first test while logged into site B, and it correctly switched to site A and took me to the order. But then I tried to switch back to site B and it took me to that site A order again.

**Update:** Perhaps this is due to using a debug build? I was unable to build the release variant, so perhaps you can DM me the APK?

**Comment by nbradbury** [comment]
@wzieba For test #5, I'm consistently taken to an empty order screen.

**Comment by wzieba** [comment]
Thank you for the review @nbradbury !

> I tried the first test while logged into site B, and it correctly switched to site A and took me to the order. But then I tried to switch back to site B and it took me to that site A order again.

I've tried reproducing it but couldn't (or maybe I'm doing something different?)

https://user-images.githubusercontent.com/5845095/193873745-ae9ca30f-0669-4745-a1bb-a9bc473ec123.mov

> For test https://github.com/woocommerce/woocommerce-android/issues/5, I'm consistently taken to an empty order screen.

Sorry, I got too "creative" when writing tests 🤦 . It's an invalid case.

It works as it should - we don't check in the database if an Order with id from URL parameters exists. There's no need - the id is provided from the core. Even if Order not yet fetched but exists in the core, it will be fetched after opening the screen.

**Comment by nbradbury** [comment]
@wzieba I can still repro this in the release build. Here's a video so you see the steps:

[device-2022-10-04-131837.webm](https://user-images.githubusercontent.com/3903757/193885390-706fd895-78d4-41e6-9b06-cb0665fba413.webm)

**Comment by wzieba** [comment]
Gotcha, @nbradbury, I misunderstood your first comment. I can reproduce the issue now, thank you for the video. I'll work on the fix and ping you.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt [hunk 1]
```diff
@@ -284,6 +285,8 @@ class MainActivity :
         if (!BuildConfig.DEBUG) {
             checkForAppUpdates()
         }
+
+        viewModel.handleIncomingAppLink(intent?.data)
```
Comment (nbradbury) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt:288]: Would it make more sense to use this?

```
intent?.data?.let {
    viewModel.handleIncomingAppLink(it)
}
```

Also, I'm wondering if this is the source of the problem I reported. Do we need to nullify the intent data here?

**Comment by wzieba** [comment]
I've found the issue with tasks + nullified the intent data as you suggested.

Screenshot 2022-10-04 at 22 42 41

Before the change, WooCommerce activity was launched in scope from which deep link was invoked. So in the sample above, in the scope of Keep Notes, next to the "original" instance of just WooCommerce. 

After 4d7f188, WooCommerce will always be invoked in its own task. If the task is running, we'll refer to active instance (hence I had to edit `onNewIntent` - before, every time Android created a new activity so `onCreate` was called).

But this change led to the discovery of another bug, which I described in a separate issue as it also relates to Notifications: #7483 . This makes 1 and 2 tests not valid anymore, unless every time after opening order details, you'll close them before running a new deep link.

----

Overall, I think that PR could be merged, even with this bug, as it's not critical IMO. I'll work on a fix in a separate PR, and I believe it can land even in another release. I think that the basic "open order details" case could work fine now. What's your opinion on that @nbradbury ?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt (hunk 1 above)
Comment (wzieba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt:288]: This check is done [here](https://github.com/woocommerce/woocommerce-android/pull/7386/files#diff-83ceec201bb8cd6d0a84d4126eb395a66325ebc48d1a0ada8ceb70ae93b6b391R20) but you pointed right with 
>  Do we need to nullify the intent data here?

I added it [here](https://github.com/woocommerce/woocommerce-android/pull/7386/files#diff-9f23f9905cef940037fc3838db74851ee987cb0fda254dccac280ce69d849a19R708). Thanks!

**Review by nbradbury — APPROVED** [review]
Looks good, thanks for the improvements! :shipit:

**Pull request merged**